[CI] EsqlClientYamlIT test {p0=esql/70_locale/Date format with default locale} failing

**Build Scans:**
- [elasticsearch-intake #10160 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mnpukxc2eyujq)
- [elasticsearch-pull-request #32906 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/p2d2vszsm7ila)
- [elasticsearch-pull-request #32889 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/7qezvskmmd2b2)
- [elasticsearch-intake #10157 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lnot53iegwnf6)
- [elasticsearch-pull-request #32900 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/74zj7pe24dxyg)
- [elasticsearch-pull-request #32135 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mlqv4mhk5z2du)
- [elasticsearch-pull-request #32114 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/gmz6glkjl2pgq)
- [elasticsearch-pull-request #32088 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2matpssnudnd4)
- [elasticsearch-pull-request #32088 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ksxvdds4ou4y6)
- [elasticsearch-pull-request #32051 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/hji6qhogsvauu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.16.0#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/70_locale/Date format with default locale}" -Dtests.seed=4FCEC800A0DA3B0C -Dtests.bwc=true -Dtests.locale=ga -Dtests.timezone=Africa/Mbabane -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Desql%2F70_locale%2FDate%20format%20with%20default%20locale%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/70_locale:29]: got unexpected warning header [
	299 Elasticsearch-8.16.0-f7911fedf5e5c6e5d645c45805d0d18d1dae53c8 "Date format [MMMM] contains textual field specifiers that could change in JDK 23. For more information, see https://ela.st/jdk-23-locales"
]

```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {p0=esql/70_locale/Date format with default locale}
- [main] 3 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 13 failures in test test {p0=esql/70_locale/Date format with default locale} (1.4% fail rate in 946 executions)
- [main] 13 failures in step 8.16.0_bwc-snapshots (3.4% fail rate in 387 executions)
- [main] 4 failures in pipeline elasticsearch-intake (3.0% fail rate in 133 executions)
- [main] 8 failures in pipeline elasticsearch-pull-request (2.3% fail rate in 345 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -293,12 +293,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}
   issue: https://github.com/elastic/elasticsearch/issues/113537
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/70_locale/Date format with default locale}
-  issue: https://github.com/elastic/elasticsearch/issues/113539
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/70_locale/Date format with Italian locale}
-  issue: https://github.com/elastic/elasticsearch/issues/113540
 - class: org.elasticsearch.xpack.inference.TextEmbeddingCrudIT
   method: testPutE5WithTrainedModelAndInference
   issue: https://github.com/elastic/elasticsearch/issues/113565

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/70_locale.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/70_locale.yml
@@ -29,7 +29,7 @@ setup:
   - do:
       allowed_warnings_regex:
         - "No limit defined, adding default limit of \\[.*\\]"
-        - "Date format \\[MMMM\\] contains textual field specifiers that could change in JDK 23"
+        - "Date format \\[MMMM\\] contains textual field specifiers that could change in JDK 23.*"
       esql.query:
         body:
           query: 'FROM events | eval fixed_format = date_format("MMMM", @timestamp), variable_format = date_format(format, @timestamp) | sort @timestamp | keep @timestamp, fixed_format, variable_format'
@@ -51,7 +51,7 @@ setup:
   - do:
       allowed_warnings_regex:
         - "No limit defined, adding default limit of \\[.*\\]"
-        - "Date format \\[MMMM\\] contains textual field specifiers that could change in JDK 23"
+        - "Date format \\[MMMM\\] contains textual field specifiers that could change in JDK 23.*"
       esql.query:
         body:
           query: 'FROM events | eval fixed_format = date_format("MMMM", @timestamp), variable_format = date_format(format, @timestamp) | sort @timestamp | keep @timestamp, fixed_format, variable_format'

